src/Tokenizer/Analyzer/CommentsAnalyzer.php

```diff
@@ -27,6 +29,67 @@
     const TYPE_DOUBLE_SLASH = 2;
     const TYPE_SLASH_ASTERISK = 3;
 
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return bool
+     */
+    public function isHeaderComment(Tokens $tokens, $index)
+    {
+        if (!$tokens[$index]->isGivenKind([T_COMMENT, T_DOC_COMMENT])) {
+            throw new \InvalidArgumentException('Given index must point to a comment.');
+        }
+
+        $prevIndex = $tokens->getPrevMeaningfulToken($index);
+
+        return $tokens[$prevIndex]->isGivenKind([T_OPEN_TAG]) && null !== $tokens->getNextMeaningfulToken($index);
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return bool
+     */
+    public function isPhpdocCandidate(Tokens $tokens, $index)
```

Comment:
please add description what this method computes.

what is candidate? candidate for what? what is the difference between real phpdoc and candidate for it?

Reply:
Added, also I've renamed the function to better self explaining.